experimental/datasets/hugging_face_dataset.py

```diff
@@ -0,0 +1,136 @@
+import os
+import pprint
+from itertools import groupby
+from operator import itemgetter
+from pathlib import Path
+
+import srsly
+from datasets import load_dataset, ClassLabel, DatasetDict, Dataset
+from flair.data import Corpus
+
+from flair.datasets import CSVClassificationDataset
+from typing import Optional, Dict
+
+from experimental.data.io import T_path
+from experimental.datasets.base import BaseDataset
+from experimental.datasets.utils.misc import all_files_exists
+from experimental.defaults import DATASET_PATH
+from experimental.utils.loggers import get_logger
+
+HUGGING_FACE_SUBSETS = ["train", "validation", "test"]
+
+
+class HuggingFaceClassificationDataset(BaseDataset):
+    def __init__(
+        self,
+        dataset_name,
+        input_text_column_name,
+        target_column_name,
+        output_path: Optional[T_path] = None,
+    ):
+        self._logger = get_logger(str(self))
+
+        if output_path is None:
+            output_path = DATASET_PATH.joinpath("polemo2")
+
+        self.output_path: Path = (
+            Path(output_path) if output_path else DATASET_PATH.joinpath("polemo2")
+        )
+
+        self.dataset_name = dataset_name
+        self.input_text_column_name = input_text_column_name
+        self.target_column_name = target_column_name
+
+        if not all_files_exists(
+            path=self.output_path, files=[x + ".csv" for x in HUGGING_FACE_SUBSETS]
+        ):
+            os.makedirs(self.output_path, exist_ok=True)
```

Comment:
I would stick to Pathlib paths instead of os paths

Reply:
noted to do that in the next version